### PR TITLE
Preserve javadoc and ordering for Thrift IDL

### PR DESCRIPTION
**Pull request by AlekseiS**

### CODE REVIEW

File: presto-thrift-connector-api/pom.xml [hunk 1]
```diff
@@ -47,6 +47,19 @@
             <artifactId>jackson-annotations</artifactId>
         </dependency>
 
+        <!-- for preserving javadoc and ordering of generated Thrift IDL classes, needed during compilation only -->
+        <dependency>
+            <groupId>com.facebook.swift</groupId>
+            <artifactId>swift-javadoc</artifactId>
+            <scope>provided</scope>
```
Comment (dain) [inline presto-thrift-connector-api/pom.xml]: I'm not sure if this is the right scope.  @electrum?

Comment (AlekseiS) [inline presto-thrift-connector-api/pom.xml]: These jars are needed for compilation only, but are not needed at runtime. "provided" scope is the one that matches it best.

Comment (electrum) [inline presto-thrift-connector-api/pom.xml]: Yeah, this is the best that can be done

**Comment by AlekseiS** [comment]
Let me try to clarify the comment and add commit message.

**Comment by AlekseiS** [comment]
Clarified the comment in pom file and added commit description.

**Comment by electrum** [comment]
I still don't understand what's happening. How does adding dependencies do anything at all?

**Comment by AlekseiS** [comment]
Look for the last bullet point in "Tips" section in swift2thrift generator docs: https://github.com/facebook/swift/tree/master/swift2thrift-generator-cli#tips
From what I see, swift-javadoc registers a javadoc processing class which appends additional metadata for compiled swift-annotated classes. That way, when you run swift2thrift this information is added to the generated Thrift IDL.

**Comment by electrum** [comment]
Makes sense for swift-javadoc. Update the comment to mention the annotation processor.

Though I still don't understand why swift-codec is needed. Maybe it's not?

**Comment by AlekseiS** [comment]
It fails without swift-codec, because referenced classes are not found. I'll update the comment.

**Comment by AlekseiS** [comment]
Updated the comment in pom file.

**Comment by AlekseiS** [comment]
Rebased and merged.

**Pull request closed without merging**